src/protocolsupport/injector/pe/PEProxyNetworkManager.java

```diff
@@ -14,7 +16,16 @@
 	protected Channel serverconnection;
 
 	@Override
-	protected void channelRead0(ChannelHandlerContext ctx, ByteBuf bytebuf) throws Exception {
+	public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception {
+		if (msg instanceof RakNetServer.BackPressure && serverconnection != null) {
+			serverconnection.config().setAutoRead(RakNetServer.BackPressure.OFF.equals(msg));
```

Comment:
What's this?

Reply:
My new raknet branch has back-pressure signals (stops reading from the downstream TCP connection when the raknet frame queue hits a high watermark). keeps bungeecord from filling the heap unnecessarily when a client is a slow connection

Reply:
actually right i forgot- there is a limited form of backpressure in my master branch of RakNetServer too